okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -140,6 +149,24 @@ private static ResponseBody buffer(MediaType mediaType, byte[] bodyBytes) {
 
     @Override
     public void enqueue(Callback callback) {
+        ListenableFuture<Listener> listenerFuture = limiter.acquire();
+        Futures.addCallback(listenerFuture, new FutureCallback<Listener>() {
```

Comment:
when you're unable to acquire, this callback ends up getting called by another thread that is completing a some other request. that seems odd?

Reply:
also wondering if there's some way for a request to be stuck in the enqueued state and never have it's Listener get set because there's no available slots when all in flight requests complete and then no new requests get enqueued so the queue is never drained

Reply:
no, becaus ethis isn't rate limiting - it's concurrency limiting